packages/libs/web-common/src/components/SiteSearch/TypeAheadInput.tsx

```diff
@@ -0,0 +1,504 @@
+  e.code === 'ArrowUp' || e.keyCode == 38;
+
+/**
+ * Tests whether the keyboard event was for an "Enter" key press.
+ *
+ * @param e Keyboard event to test.
+ *
+ * @return `true` if the keyboard event was for an "Enter" key press, otherwise
+ * `false`.
+ */
+const kbIsEnter = (e: React.KeyboardEvent<any>) =>
+  e.code === 'Enter' || e.keyCode == 13;
+
+/**
+ * Tests whether the keyboard event was for an "Escape" key press.
+ *
+ * @param e Keyboard event to test.
+ *
+ * @return `true` if the keyboard event was for an "Escape" key press, otherwise
+ * `false`.
+ */
+const kbIsEscape = (e: React.KeyboardEvent<any>) =>
+  e.code === 'Escape' || e.keyCode == 27;
+
+// endregion Keyboard
+
+// region Strings
+
+//
+// String Helper Functions
+//
+
+/**
+ * Returns the last word of the given string.
+ *
+ * @param value String from which the last word should be returned.
+ *
+ * @return The last word of the given string.
+ */
+const lastWordOf = (value: string) =>
+  ((arr: Array<string>) => arr.length > 0 ? arr[arr.length - 1] : '')(value.split(/ +/));
+
+/**
+ * Replaces the last word in the given `original` string with the given
+ * `replacement` word(s), returning the new string.
+ *
+ * @param original Original string whose last word should be replaced.
+ *
+ * @param replacement Word that will replace the last word of the original
+ * string.
+ *
+ * @return A new string formed by replacing the last word of the `original`
+ * string with the given `replacement`.
+ */
+const replaceLastWord = (original: string, replacement: string) =>
+  original.substring(0, original.lastIndexOf(lastWordOf(original))) +
+  replacement;
+
+// endregion Strings
+
+// region Debouncer
+
+/**
+ * Debouncer Function Type.
+ *
+ * Represents a function that may be used to build a "debounced" or "debouncing"
+ * function by passing in a target function that takes a value of type `T` and
+ * returns nothing.
+ *
+ * The return value of a Debouncer function call will be a new function that
+ * wraps the given target function with debouncing.
+ *
+ * @param T Type of the value consumed by the function wrapped by and returned
+ * by the Debouncer function.
+ */
+type Debouncer<T> = (func: (value: T) => void) => (value: T) => void;
+
+
+/**
+ * Builds a new `Debouncer` function that may be used to build one or more
+ * functions that debounce on the same timer.
+ *
+ * @param delay Debouncing delay.
+ *
+ * @return A `Debouncer` function that may be used to build one or more
+ * functions that debounce on the same timer.
+ */
+function buildDebouncer<T>(delay: number): Debouncer<T> {
+  let timer: any;
+
+  return (func: (value: T) => any) => {
+    return (value: T) => {
+      clearTimeout(timer);
+      timer = setTimeout(() => func(value), delay);
+    };
+  };
+}
+
+// endregion Debouncer
+
+// region TypeAheadAPI
+
+const TYPEAHEAD_PATH: string = 'suggest';
+
+/**
+ * Wrapper for the SiteSearch Type-Ahead HTTP API.
+ */
+class TypeAheadAPI {
```

Comment:
See https://github.com/VEuPathDB/web-monorepo/blob/main/packages/libs/http-utils/src/FetchClient.ts, for a base class you can use to implement this. Note that you can use it with https://github.com/VEuPathDB/web-monorepo/blob/main/packages/libs/http-utils/src/ioTransformer.ts to build run-time validation.

See an example of how it's used here: https://github.com/VEuPathDB/web-monorepo/blob/main/packages/libs/eda/src/lib/core/api/SubsettingClient/index.ts.

Also, note that API methods return a `Promise<T>`, rather than taking a callback, so you would also have to modify the caller of `typeahead`, if you go this route.